modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZonesUtil.java

```diff
@@ -526,26 +533,71 @@ public static boolean filter(Map<String, String> nodeAttributes, String filter)
         return !res.isEmpty();
     }
 
+    /**
+     * Filters storage profiles.
+     *
+     * @param node Node with storage profile attributes.
+     * @param zoneStorageProfiles Zone's storage profiles.
+     * @return True, if matches, false otherwise.
+     */
+    public static boolean filterStorageProfiles(
+            NodeWithAttributes node,
+            List<CatalogStorageProfileDescriptor> zoneStorageProfiles
+    ) {
+        if (node.storageProfiles() == null) {
+            return false;
+        }
+
+        List<String> zoneStorageProfilesNames = zoneStorageProfiles.stream()
+                .map(CatalogStorageProfileDescriptor::storageProfile)
+                .collect(Collectors.toList());
+
+        // TODO:
```

Comment:
TODO without ticket. But anyway, I think we shouldn't make the special case for dummy or default profiles here. Instead we should avoid creation of zone with empty list of profiles.